File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cs
public class Solution {
    public int MinLength(string s) {
        var stack = new Stack<char>();
        foreach(var ch in s){
            if(stack.Count == 0)
                stack.Push(ch);
            else if(stack.Peek() == 'A' && ch == 'B')
                stack.Pop();
            else if(stack.Peek() == 'C' && ch == 'D')
                stack.Pop();
            else
                stack.Push(ch);
        }
        return stack.Count;
    }
}

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        for ch in s:
            if not stack:
                stack.append(ch)
            elif stack[-1] == 'A' and ch == 'B':
                stack.pop()
            elif stack[-1] == 'C' and ch == 'D':
                stack.pop()
            else:
                stack.append(ch)
        return len(stack)